### PR TITLE
Replaces rights with a static groups system

### PR DESCRIPTION
**Pull request by vmarseguerra**
## Converts the rights + groups system to a groups only system

The change was made to resolve the difference between the front and the api:
The front only has groups no concept of rights and base the UI layout on it. So after a change of rights in the database, the front UI is broken until it is adapted. Also changing the front to use a dynamic right system was not worth the effort given how rarely the right needs to be changed. So the API was adapted to use the same groups only system which is simpler and so easier to maintain. However now, if a change of right is wanted a code change is required in the API and in the front.

Groups have no inheritance or order. They can be seen as only providing additional rights to a user. Users can be a member of no, one or multiple groups. Rights for each endpoints should not have changed, only the system that enforces them has changed.
As a reminder, here is a summary of what each groups allows:
- Administrator:
    - Allow to change group membership of any users
    - Can remove the sensitive flag on entrances
    - Can make bulk CSV import of entrances and documents
- Moderator:
    - Can remove entities
    - Can validate documents
    - Can handle document and entrance duplicate
    - Can edit any comments
- Leader:
    - Can subscribe to massifs and countries
    - Can download the DB export
   
Groups are also used to set the allowed requests rate limit, see #1157 (implemented in #1223) for details

Like before, the requirements for a route to be accessed by anyone or only logged users is done using sails policy. To avoid redundancy only the `tokenAuth` policy has been kept, verification of group membership is done in the code of each controllers.
Roughly, visitors can only get entities data while logged users can also create and update them.

In the database, the old `Visitor` and `User` groups are no longer used and have been removed.

To ease the review, reviewers can you this SQL query to retrieve all previous rights associated with their groups:
```
SELECT t_group.name, t_right.name
FROM t_group
JOIN j_group_right ON t_group.id = j_group_right.id_group
JOIN t_right ON t_right.id = j_group_right.id_right
ORDER BY t_group.name, t_right.name;
```

Once this PR is merged and deployed to prod, the following queries can be run to clean up the production database:
```
DELETE FROM t_group WHERE id = 3 OR id = 4;
DELETE FROM j_caver_group WHERE id_group = 3 OR id_group = 4;
DROP TABLE t_right;
DROP TABLE j_group_right;
```

### CODE REVIEW

**Comment by urien** [comment]
Super clair.
Les Leaders ont un rate limit supérieur a celui des Users mais inférieur aux modérateurs. de manière a leur offrir un service supplémentaire
Les modérateurs peuvent aussi faire l'import en masse via CSV de documents et d'entrées. Je pense qu'il serait préférable de garder cette fonctionnalité aux admins car c'est un outil qui peut avoir un impact assez important sur la base

**Comment by Clm-Roig** [comment]
@vmarseguerra The production database rights: 

|Group Name|Right name|
|----|----|
|Administrator|Application - delete any|
|Administrator|Application - edit any|
|Administrator|Application - no moderator delete request limit|
|Administrator|Application - no request limit|
|Administrator|Application - no user delete request limit|
|Administrator|Application - view any|
|Administrator|Caver - delete any|
|Administrator|Caver - edit any|
|Administrator|Comment - edit any|
|Administrator|Document - delete any|
|Administrator|Entrance - unmark as sensitive|
|Administrator|Entrance - view complete|
|Administrator|Group - delete any|
|Administrator|Group - edit any|
|Administrator|Group - view any|
|Administrator|Right - delete any|
|Administrator|Right - edit any|
|Administrator|Right - view any|
|Moderator|Application - no request limit|
|Moderator|Application - no user delete request limit|
|Moderator|Application - view any|
|Moderator|Cave - delete any|
|Moderator|Cave - unlink resource|
|Moderator|Comment - edit any|
|Moderator|Document - csv import|
|Moderator|Document Duplicate - create|
|Moderator|Document Duplicate - delete any|
|Moderator|Document - edit not validated|
|Moderator|Entrance - csv import|
|Moderator|Entrance - delete any|
|Moderator|Entrance Duplicate - create|
|Moderator|Entrance Duplicate - delete any|
|Moderator|Entrance - unlink resource|
|Moderator|Group - view any|
|Moderator|Massif - delete any|
|Moderator|Organization - delete any|
|Moderator|Right - view any|
|User|Application - no request limit|
|User|Cave - create|
|User|Cave - edit any|
|User|Cave - link resource|
|User|Cave - merge|
|User|Caver - create|
|User|Caver - delete own|
|User|Caver - edit own|
|User|Caver - view any|
|User|Caver - view complete|
|User|Cave - view any|
|User|Comment - create|
|User|Comment - edit own|
|User|Country - subscribe|
|User|Description - create|
|User|Description - edit any|
|User|Document - create|
|User|Document - edit any|
|User|Document - view any|
|User|Entrance - create|
|User|Entrance - edit any|
|User|Entrance - link resource|
|User|Entrance - mark as sensitive|
|User|Entrance - view limited|
|User|History - create|
|User|History - edit any|
|User|Location - create|
|User|Location - edit any|
|User|Massif - create|
|User|Massif - edit any|
|User|Massif - subscribe|
|User|Massif - view any|
|User|Name - edit any|
|User|Organization - create|
|User|Organization - edit any|
|User|Organization - view any|
|User|Rigging - create|
|User|Rigging - edit any|
|Visitor|Caver - view any|
|Visitor|Document - view any|

**Comment by Clm-Roig** [comment]
I'm completely changing the rate limit functions in #1223. I will adapt my code to use yours (instead of the old ones) of course.

**Comment by vmarseguerra** [comment]
@urien  C'est modifié, les imports de CSV sont maintenant réservés aux administrateurs

@Clm-Roig Ok top, j'ai modifié la description pour rediriger vers l'issue et le PR qui modifie les rates limit

**Comment by urien** [comment]
Merci beaucoup 👍

File: api/responses/forbidden.js [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * @param  {Object | string | undefined} data
  */
 
-module.exports = function forbidden(data) {
+module.exports = function forbidden(data = 'Request forbidden') {
```
Comment (Clm-Roig) [inline api/responses/forbidden.js:16]: 'Forbidden request'

File: sql/2_2021_09_22_1_groups.sql
Comment (Clm-Roig) [inline sql/2_2021_09_22_1_groups.sql:1]: Visitor group can be deleted I think

File: test/integration/4_routes/Caves/delete.test.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const AuthTokenService = require('../../AuthTokenService');
 describe('Cave features', () => {
   let adminToken;
   before(async () => {
-    adminToken = await AuthTokenService.getRawBearerAdminToken();
+    adminToken = await AuthTokenService.getRawBearerModeratorToken();
```
Comment (Clm-Roig) [inline test/integration/4_routes/Caves/delete.test.js:8]: Something is wrong here. Either rename the variable or change the token type no ?

File: test/integration/4_routes/Comments/update.test.js [hunk 3]
```diff
@@ -4,10 +4,12 @@ const AuthTokenService = require('../../AuthTokenService');
 
 describe('Comment features', () => {
   let userToken;
+  let userTokenModerator;
```
Comment (Clm-Roig) [inline test/integration/4_routes/Comments/update.test.js:7]: `moderatorToken` instead ?

File: api/controllers/v1/comment/update.js [hunk 4]
```diff
@@ -22,22 +20,17 @@ module.exports = async (req, res) => {
...
+      const hasRight = RightService.hasGroup(
+        req.token.groups,
+        RightService.G.MODERATOR
       );
-    if (!hasRight) {
-      return res.forbidden('You are not authorized to update any/own comment.');
+
+      if (!hasRight) {
+        return res.forbidden(
+          'You are not authorized to update any/own comment.'
```
Comment (Clm-Roig) [inline api/controllers/v1/comment/update.js:31]: `You are not authorized to update **any** comment.` dans ce if il me semble.

File: api/controllers/v1/document-duplicate/find-all.js [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
 const { toDocumentDuplicate } = require('../../../services/mapping/converters');
 
 module.exports = async (req, res) => {
+  const hasRight = RightService.hasGroup(
+    req.token.groups,
+    RightService.G.MODERATOR
+  );
+  if (!hasRight) {
+    return res.forbidden(
+      'You are not authorized to find all documents duplicate.'
```
Comment (Clm-Roig) [inline api/controllers/v1/document-duplicate/find-all.js:14]: `document duplicates`

File: api/controllers/v1/entrance/update.js [hunk 6]
```diff
@@ -46,26 +31,17 @@ module.exports = async (req, res) => {
     const newIsSensitiveValue = cleanedData.isSensitive;
     if (
       newIsSensitiveValue !== undefined &&
-      newIsSensitiveValue !== currentEntrance.isSensitive
+      newIsSensitiveValue !== currentEntrance.isSensitive &&
+      !newIsSensitiveValue
```
Comment (Clm-Roig) [inline api/controllers/v1/entrance/update.js:35]: Ici tu veux dire `newIsSensitiveValue === false` si j'ai bien compris ? Je trouve ça plus clair plutôt que de simplement mettre `!newIsSensitiveValue`.

**Review by Clm-Roig — CHANGES_REQUESTED** [review]
Je refuse juste pour des changements mineurs assez peu impactants au final.

Mais sinon super boulot !

**Review by Clm-Roig — APPROVED** [review]

**Pull request merged**